flocker/node/agents/ebs.py

```diff
@@ -682,13 +785,43 @@ def _next_device(self, instance_id, volumes, devices_in_use):
 
     def create_volume(self, dataset_id, size):
         """
+        Create a volume on EBS backend.
+        """
+        return self.create_volume_with_profile(
+            dataset_id, size, MandatoryProfiles.DEFAULT.value)
+
+    def create_volume_with_profile(self, dataset_id, size, profile_name):
+        """
         Create a volume on EBS. Store Flocker-specific
         {metadata version, cluster id, dataset id} for the volume
         as volume tag data.
         Open issues: https://clusterhq.atlassian.net/browse/FLOC-1792
         """
-        requested_volume = self.connection.create_volume(
-            size=int(Byte(size).to_GiB().value), zone=self.zone)
+        requested_size = int(Byte(size).to_GiB().value)
+        try:
+            volume_type, iops = _volume_type_and_iops_for_profile_name(
+                profile_name, requested_size)
+            requested_volume = self.connection.create_volume(
+                size=requested_size,
+                zone=self.zone,
+                volume_type=volume_type,
+                iops=iops)
+        except EC2ResponseError as e:
+            # If we failed to create a volume with attributes complying
+            # with requested profile, make a second attempt at volume
+            # creation with default profile.
+            # Compliance violation of the volume's requested profile
+            # will be detected and remediated in a future release (FLOC-3275).
```

Comment:
This will result in volumes being created with wrong profile due to unrelated intermittent errors, e.g. rate limit exceeded. So it would be good to at least check for a specific EC2 error indicating "couldn't create with these options" rather than doing this unconditionally for all errors. That could be done in separate issue if you want.